core/trino-main/src/main/java/io/trino/server/security/oauth2/ScribeHttpClient.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server.security.oauth2;
+
+import com.github.scribejava.core.httpclient.HttpClient;
+import com.github.scribejava.core.httpclient.multipart.MultipartPayload;
+import com.github.scribejava.core.model.OAuthAsyncRequestCallback;
+import com.github.scribejava.core.model.OAuthRequest.ResponseConverter;
+import com.github.scribejava.core.model.Response;
+import com.github.scribejava.core.model.Verb;
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.io.CharStreams;
+import io.airlift.http.client.BodyGenerator;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Request;
+import io.airlift.http.client.ResponseHandler;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.airlift.http.client.StaticBodyGenerator.createStaticBodyGenerator;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.USER_AGENT;
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+class ScribeHttpClient
+        implements HttpClient
+{
+    private static final Multimap<String, String> DEFAULT_HEADERS = ImmutableMultimap.of(CONTENT_TYPE, APPLICATION_FORM_URLENCODED, ACCEPT, APPLICATION_JSON);
+
+    private final io.airlift.http.client.HttpClient httpClient;
+
+    ScribeHttpClient(io.airlift.http.client.HttpClient httpClient)
+    {
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+    }
+
+    @Override
+    public <T> Future<T> executeAsync(
+            String userAgent,
+            Map<String, String> headers,
+            Verb httpVerb,
+            String completeUrl,
+            byte[] bodyContents,
+            OAuthAsyncRequestCallback<T> callback,
+            ResponseConverter<T> converter)
+    {
+        throw new UnsupportedOperationException("Not implemented");
```

Comment:
Can you please create some base class with all methods throwing `UnsupportedOperationException`